pkg/iac/terraform/state/backend/tfcloud_reader.go

```diff
@@ -28,72 +31,93 @@ type TFCloudBody struct {
-	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/workspaces/%s/current-state-version", opts.TFCloudEndpoint, workspaceId), nil)
+func NewTFCloudReader(client pkghttp.HTTPClient, workspacePath string, opts *Options) (*TFCloudBackend, error) {
+	token, err := GetToken(opts)
 	if err != nil {
 		return nil, err
 	}
-	req.Header.Add("Content-Type", "application/vnd.api+json")
-	return &TFCloudBackend{req, client, nil, opts}, nil
+	config := &tfe.Config{
+		Token:      token,
+		Address:    opts.TFCloudEndpoint,
+		HTTPClient: client.(*http.Client),
+	}
+	tfcClient, err := tfe.NewClient(config)
+	if err != nil {
+		return nil, err
+	}
+	return &TFCloudBackend{tfcClient, nil, opts, workspacePath}, nil
 }
 
-func (t *TFCloudBackend) authorize() error {
-	token := t.opts.TFCloudToken
+func GetToken(opts *Options) (string, error) {
+	token := opts.TFCloudToken
 	if token == "" {
 		tfConfigFile, err := getTerraformConfigFile()
 		if err != nil {
-			return err
+			return "", err
 		}
+
 		file, err := os.Open(tfConfigFile)
 		if err != nil {
-			return err
+			return "", err
 		}
 		defer file.Close()
 		reader := NewTFCloudConfigReader(file)
-		token, err = reader.GetToken(t.request.URL.Host)
+
+		u, err := url.Parse(opts.TFCloudEndpoint)
 		if err != nil {
-			return err
+			return "", err
 		}
+		return reader.GetToken(u.Host)
+	}
+	return token, nil
+}
+
+// A regular expression used to validate string workspace ID patterns.
+var reStringID = regexp.MustCompile(`^ws-[a-zA-Z0-9\-\._]+$`)
+
+// isValidWorkspaceID checks if the given input is present and non-empty.
+func isValidWorkspaceID(v string) bool {
+	return v != "" && reStringID.MatchString(v)
+}
+
+func (t *TFCloudBackend) GetWorkspaceId() (string, error) {
+	if isValidWorkspaceID(t.workspacePath) {
+		return t.workspacePath, nil
+	}
+	workspacePath := strings.Split(t.workspacePath, "/")
+	if len(workspacePath) != 2 {
+		return "", fmt.Errorf("invalid workspace path")
+	}
+	workspace, err := t.client.Workspaces.Read(context.Background(), workspacePath[0], workspacePath[1])
+	if err != nil {
+		return "", err
 	}
-	t.request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
-	return nil
+	return workspace.ID, nil
 }
 
 func (t *TFCloudBackend) Read(p []byte) (n int, err error) {
 	if t.reader == nil {
-		if err := t.authorize(); err != nil {
-			return 0, err
-		}
-		res, err := t.client.Do(t.request)
+		workspaceId, err := t.GetWorkspaceId()
 		if err != nil {
 			return 0, err
 		}
 
-		if res.StatusCode < 200 || res.StatusCode >= 400 {
-			return 0, errors.Errorf("error requesting terraform cloud backend state: status code: %d", res.StatusCode)
-		}
-
-		body := TFCloudBody{}
-		bodyBytes, _ := ioutil.ReadAll(res.Body)
-		err = json.Unmarshal(bodyBytes, &body)
+		stateVersion, err := t.client.StateVersions.Current(context.Background(), workspaceId)
 		if err != nil {
 			return 0, err
```

Comment:
Not sure what the error looks like, I was wondering if we should wrap it to give the user more context on what failed ?
like
```go
return 0,errors.WithMessage("error requesting terraform cloud backend state", err)
```